test/unit/authorizer.test.js

```diff
@@ -28,52 +29,56 @@ describe('Authorizers', function () {
         it('Auth header must be added', function () {
             var request = new Request(rawRequests.basic),
                 auth = request.auth,
+                username = rawRequests.basic.auth.basic.username,
+                password = rawRequests.basic.auth.basic.password,
+                expectedAuthHeader = 'Authorization: Basic ' + btoa(username + ':' + password),
                 handler = Authorizer.Handlers[auth.type],
                 authorizedReq = handler.sign(auth.parameters().toObject(), request),
                 headers = authorizedReq.headers.all(),
                 authHeader;
             expect(headers.length).to.eql(1);
 
             authHeader = headers[0];
-            expect(authHeader.toString()).to.eql('Authorization: Basic YWJoaWppdDprYW5l');
+            expect(authHeader.toString()).to.eql(expectedAuthHeader);
             expect(authHeader.system).to.be(true);
         });
 
-        it('should bail out if username is missing', function () {
```

Comment:
now if something is missing we make default it to empty string

Reply:
You could retain the test and assert that the fallback is indeed an empty string. (If not done already)

Reply:
that is handled in the test case `should use default values for the missing parameters`

Reply:
line 80